### PR TITLE
Create container

### PR DESCRIPTION
**Pull request by rickle-msft**

### CODE REVIEW

**Review by gapra-msft — APPROVED** [review]

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/android/storage/blob/StorageBlobServiceImpl.java [hunk 1]
```diff
@@ -1556,6 +1693,17 @@ private static String readAsString(ResponseBody body) {
     }
 
     private interface StorageBlobService {
+        @PUT("{containerName}")
+        Call<ResponseBody> createContainer(@Path("containerName") String containerName,
+                                                      @Query("timeout") Integer timeout,
+                                                      @Header("x-ms-meta-") Map<String, String> metadata,
```
[the remark this answers is not in the document]
Comment (rickle-msft) [reply sdk/storage/azure-storage-blob/src/main/java/com/azure/android/storage/blob/StorageBlobServiceImpl.java:2033]: Yea I figured it was a long shot that retrofit supported our weird metadata prefix situation :). I kinda put that there as a place holder until we got some live tests going and could see a little more info. Now I'm there and have confirmed it doesn't work! I'll give your suggestion a shot and see where that gets us! Thanks for the head start!

[the remark this answers is not in the document]
Comment (rickle-msft) [reply sdk/storage/azure-storage-blob/src/main/java/com/azure/android/storage/blob/StorageBlobServiceImpl.java]: Should be fixed

**Review by gapra-msft — APPROVED** [review]

File: sdk/core/azure-core/src/test/java/com/azure/android/core/common/TestUtils.java [hunk 2]
```diff
@@ -1,29 +1,141 @@
...
+            .setCredentialInterceptor(new SasTokenCredentialInterceptor(new SasTokenCredential(getDefaultSasToken())));
+    }
+
+    public static StorageBlobAsyncClient.Builder initializeDefaultAsyncBlobClientBuilder(Interceptor ... interceptors) {
+        return initializeDefaultAsyncBlobClientBuilder(enableFiddler(), interceptors);
+    }
+
+    public static StorageBlobAsyncClient.Builder initializeDefaultAsyncBlobClientBuilder(boolean enableFiddler,
+                                                                                         Interceptor ... interceptors) {
+        return new StorageBlobAsyncClient.Builder(UUID.randomUUID().toString(),
```
Comment (gapra-msft) [inline sdk/core/azure-core/src/test/java/com/azure/android/core/common/TestUtils.java:64]: can we replace all instances of UUID.randomUUID() with a method call? That way when we get to the point where we record, we can just change that method?

Comment (gapra-msft) [reply sdk/core/azure-core/src/test/java/com/azure/android/core/common/TestUtils.java:64]: Same idea with date time now

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/android/storage/blob/models/ContainerGetPropertiesHeaders.java [hunk 3]
```diff
@@ -0,0 +1,506 @@
...
+        this.denyEncryptionScopeOverride = denyEncryptionScopeOverride;
+        return this;
+    }
+
+    /**
+     * Get the errorCode property: The errorCode property.
+     *
+     * @return the errorCode value.
+     */
+    public String getErrorCode() {
```
Comment (gapra-msft) [inline sdk/storage/azure-storage-blob/src/main/java/com/azure/android/storage/blob/models/ContainerGetPropertiesHeaders.java:492]: TODO: Remove error code

File: sdk/storage/azure-storage-blob/src/test/java/com/azure/android/storage/blob/ContainerTest.java [hunk 4]
```diff
@@ -0,0 +1,368 @@
+package com.azure.android.storage.blob;
```
Comment (gapra-msft) [inline sdk/storage/azure-storage-blob/src/test/java/com/azure/android/storage/blob/ContainerTest.java:1]: Add Copyright header?

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/android/storage/blob/models/ContainerGetPropertiesHeaders.java (hunk 3 above)
Comment (rickle-msft) [reply sdk/storage/azure-storage-blob/src/main/java/com/azure/android/storage/blob/models/ContainerGetPropertiesHeaders.java:492]: Done

File: sdk/storage/azure-storage-blob/src/test/java/com/azure/android/storage/blob/ContainerTest.java (hunk 4 above)
Comment (rickle-msft) [reply sdk/storage/azure-storage-blob/src/test/java/com/azure/android/storage/blob/ContainerTest.java:1]: For some reason it looked like none of the tests had these, so I just followed that, but I'll add it to the other test files, too.

File: sdk/core/azure-core/src/test/java/com/azure/android/core/common/TestUtils.java (hunk 2 above)
Comment (rickle-msft) [reply sdk/core/azure-core/src/test/java/com/azure/android/core/common/TestUtils.java]: Done

**Pull request merged**